### ISSUE TITLE
Redefination of layouts for screens of different sizes and for landscape mode

### ISSUE THREAD

**Issue by Rimjhim28** [systers/volunteers-android#88]
Currently, the app uses a single defination of layouts for portrait and landscape modes. Also, the same layout is used for different screen sizes. This is resulting in just stretching of the UI elements to fit the screen. Instead, we can create different layouts for landscape and different screen sizes to have a proper orientation of the elements and for better utilization of the emerging white spaces.

For instance, redefination of layouts for tablet to have a grid view instead of a list view could be better option. Currently, the usage of listview results in unused white spaces for tablets.
![screenshot_1509693641](https://user-images.githubusercontent.com/26908195/32363930-d7c3e1f0-c097-11e7-8d29-f43e5de202ab.png)
![screenshot_1509693650](https://user-images.githubusercontent.com/26908195/32363934-d9924ddc-c097-11e7-80b9-a40d6b438533.png)
![screenshot_1509693760](https://user-images.githubusercontent.com/26908195/32364023-82b9d8b2-c098-11e7-9da4-a3bd457af805.png)
![screenshot_1509693766](https://user-images.githubusercontent.com/26908195/32364028-866dec8c-c098-11e7-8d74-b42a1893ff70.png)
![screenshot_1509693801](https://user-images.githubusercontent.com/26908195/32364033-8bc2439a-c098-11e7-9355-ec7cb02388c9.png)
![screenshot_1509693812](https://user-images.githubusercontent.com/26908195/32364037-900f0b22-c098-11e7-93f4-55a660d81a4c.png)
 
Also redefining the layouts for landscape mode for devices would be better option.
Including th
![screenshot_1509694161](https://user-images.githubusercontent.com/26908195/32364093-eed9caf2-c098-11e7-84a0-32bb93c1b2a4.png)
e screenshot for landscape mode.
![screenshot_1509694170](https://user-images.githubusercontent.com/26908195/32364092-ebc0c1cc-c098-11e7-9d82-126f2370cc8f.png)

@sandarumk @HimanshiJain @pandilladivya I would like to work on this.

**Comment by pandilladivya** [comment]
@Rimjhim28 I don't think changing the UI from Listview to Gridview for bigger screens / landscape could be a better option because, there will be inconsistency in the user interface.

@sandarumk @HimanshiJain what do you suggest?

**Comment by Rimjhim28** [comment]
@pandilladivya That was just an example I used to explain the issue. Surely there can be a much better implementation for the layouts.

**Comment by pandilladivya** [comment]
@Rimjhim28 Can you suggest few by making simple mocks?

**Comment by Rimjhim28** [comment]
Ya sure! Will need some time :)

**Comment by Rimjhim28** [comment]
Here are some mocks @pandilladivya 
According to material design, the navigation drawer can always be visible for screens of larger widths.
![screenshot_1509876069](https://user-images.githubusercontent.com/26908195/32413869-5ff7f1f2-c240-11e7-8ce0-636b42cb921f.png)

Instead of having just one streched out element two elements could be a better option.
![screenshot_1509876115](https://user-images.githubusercontent.com/26908195/32413885-898b3808-c240-11e7-9fe7-3b2dc07467e5.png)

The profile activity seems a bit streched out for larger screens. Usage of left and right margins would seem better.
![screenshot_1509876139](https://user-images.githubusercontent.com/26908195/32413903-f518ebec-c240-11e7-972b-596668ebad90.png)

![screenshot_1509876149](https://user-images.githubusercontent.com/26908195/32413907-f960933a-c240-11e7-90cd-2dfc09b875ff.png)

**Comment by pandilladivya** [comment]
@Rimjhim28 Looks good to me.

Lets wait for @sandarumk @HimanshiJain to put in their views! :)

**Comment by sandarumk** [comment]
For me, I believe the change/suggestion would be a very large change and before that we do have to have a better understanding on the direction of the project as well as how much change we need until the next project lifecycle. @MeepyMay ca we have a quick meeting on the same?

**Comment by Rimjhim28** [comment]
@pandilladivya @sandarumk Should I continue with other layouts or wait for @MeepyMay 's reply.

**Comment by MeepyMay** [comment]
@sandarumk I think a meeting discussing project's lifecycle/future would be best. 

For me, I understand the "unnecessary white space" @Rimjhim28 mentioned. However, this application was built with the minimal functional product in mind. So the white space can be filled with additional features as the project grows. 

i.e. [Plus Button with Floating Actions](https://storage.googleapis.com/material-design/publish/material_v_12/assets/0B6GnvA6rl3tYWEZGZFBuc1RxMEk/components-fab-flyouts-do.png) (Add to Calendar, Share on Social Media, etc.) or Sponsor's Logos or Affinity Community label.

**Comment by Rimjhim28** [comment]
Yeah! But some elements like a permanent opened navigation drawer and left and right margins for profilr and registration in tablet views can be included. These are in accordance with the material design guidelines and would give a better view of the app. Can I work and send PRs for them? @pandilladivya @sandarumk @MeepyMay @HimanshiJain

**Comment by Rimjhim28** [comment]
@m-murad Can I send a PR for this?

**Comment by roopalJazz** [comment]
@haroonshah2764 have a look at this issue.
